db/wrapper.go

```diff
@@ -59,26 +63,33 @@ func (s *sessionWrapper) Close() {
 type databaseWrapper struct {
 	ctx      context.Context
 	database *mongo.Database
+	maxTime  *time.Duration
 }
 
 func (d *databaseWrapper) Name() string        { return d.database.Name() }
 func (d *databaseWrapper) DropDatabase() error { return errors.WithStack(d.database.Drop(d.ctx)) }
 func (d *databaseWrapper) C(coll string) Collection {
 	return &collectionWrapper{
-		ctx:  d.ctx,
-		coll: d.database.Collection(coll),
+		ctx:     d.ctx,
+		coll:    d.database.Collection(coll),
+		maxTime: d.maxTime,
```

Comment:
I would avoid sharing pointers like this between different objects. Shared pointers in separate structs when you aren't sure of the lifetime is a big no-no in pointer code. If different structs need to share the value, they can just use values instead of a shared pointer.

Reply:
this has gone away